extensions/automation/join2create.py

```diff
@@ -54,15 +451,49 @@ async def on_join2create(self, member: discord.Member, before: discord.VoiceStat
                     ),
                     member: discord.PermissionOverwrite(move_members=True),
                 }
-                channel = await after.channel.category.create_voice_channel(
-                    name=f"{member.display_name} {choice(self.suffixes)}",
-                    user_limit=25,
-                    reason="Join2Create",
-                    overwrites=perms,
-                )
-                await self.client.db.create_temp_voice(channel, member)
+                if member.guild.premium_subscriber_role:
+                    perms[member.guild.premium_subscriber_role] = discord.PermissionOverwrite(move_members=True)
+                if member.guild.default_role != role:
+                    perms[member.guild.default_role] = discord.PermissionOverwrite(view_channel=False)
+                else:
+                    perms[member.guild.default_role] = discord.PermissionOverwrite(view_channel=True)
+                channel_name = choice(self.suffixes)
+                member_name = member.display_name
+                if member_name[-1].lower() != "s":
+                    if self.status[channel_name]["s_flag"]:
+                        member_name += "s"
+
+                try:
+                    channel = await after.channel.category.create_voice_channel(
+                        name=f"{member_name}{channel_name}",
+                        user_limit=25,
+                        reason="Join2Create",
+                        overwrites=perms,
+                    )
+                except (discord.Forbidden, discord.HTTPException, discord.InvalidArgument) as e:
+                    self.logger.error(e)
+                    return
+
+                if self.status[channel_name]:
+                    status = choice(self.status[channel_name]["status"])
+                    await channel.set_status(status=status, reason="Join2Create")
+
+                check = await self.client.db.create_temp_voice(channel, member)
+                if not check:
+                    await channel.delete(reason="Join2Create-Failed")
+                    self.logger.error("Database entry for temp voice channel could not be created")
+                    return
                 self.logger.debug(f"{member.name} created {channel.name} in {member.guild}")
-                await member.move_to(channel)
+                try:
+                    await member.move_to(channel)
+                except (discord.Forbidden, discord.HTTPException) as e:
+                    self.logger.error(f"Couldn't move member | {e}")
+                    return
+                try:
+                    await channel.send(view=VoiceBoard(channel=channel))
```

Comment:
do we need to pass channel=channel is it possible to get the channel inside voice board so the bot can restart and the panel still works?